### DISCUSSION TITLE
Changing user email and signin via OTP using old email doesn't work

### DISCUSSION THREAD

**Discussion by viraj-lead** [supabase/supabase#29080] [Questions]
I use magic email signin(Signin via OTP) exclusively in my app. I also provide the option for a user to update their email address. I am using the following to update the email.
```
const { data, error } = await supabase.auth.updateUser({
                email: newEmail
            });
```

After successfully updating the email. A user cannot create another account with the old email using the SigninWithOTP
```
const { data, error } = await supabase.auth.signInWithOtp({
            email: email,
            options: {
                emailRedirectTo: destination,
            },
        });
```
The error shows up as "User is already registered".

If a user's email has already been updated, shouldn't signing in via the otp with the old email still work?

**Comment by silentworks** [comment]
If you are getting a **User is already registered** error then it means you have email confirmation turned on and you haven't confirmed the new email as yet so the old one will remain associated with the account until the new one is confirmed. I have attached a screen recording of me doing this process without the issue you are having above.

https://github.com/user-attachments/assets/11c798e7-7cee-4b2b-bfb8-d65cdc606a08
